interp/smartlocate.ml

```diff
@@ -37,8 +37,8 @@ let global_of_extended_global = function
   | TrueGlobal ref -> ref
   | SynDef kn ->
   match search_syntactic_definition kn with
-  | [],NRef ref -> ref
-  | [],NApp (NRef ref,[]) -> ref
+  | [],NRef (ref,_) -> ref
+  | [],NApp (NRef (ref,_),[]) -> ref
```

Comment:
Idem, I'm missing examples but wouldn't it be safer to put a `None` here (or else, to return the universe)?

Reply:
Returning the universe is hard, because `TrueGlobal` doesn't come with a universe instance. I wonder if these functions should take a universe instance alongside the qualified identifier, to be returned unmodified in the `TrueGlobal` case and to be substituted in the `SynDef` case once notations can have universe parameters.
(`Notation syndef@{i j} := (foo@{i} bar@{j}).` seems like good syntax for syndefs with universes.)

But that is probably out of scope for this PR; the choice here is between unexpectedly not finding notations when they have a universe instance in them, or finding notations but forgetting the universe instance, leading to unexpectedly fresh universes. The latter is closest to the current state of the world, but if you prefer the former I'm fine with that too.

Reply:
Updated to match against None here and elsewhere.

Reply:
>  `Notation syndef@{i j} := (foo@{i} bar@{j}).` seems like good syntax for syndefs with universes.

Indeed. I believe this means a 5th kind of variable to pass to notations (in addition to terms, bindings, and lists of those).

Reply:
> The latter is closest to the current state of the world, but if you prefer the former I'm fine with that too.

Not a strong opinion.